hack/generator/pkg/astmodel/package_import_set.go

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import "sort"
+
+// PackageImportSet represents a set of distinct PackageImport references
+type PackageImportSet struct {
+	imports map[PackageImport]struct{}
+}
+
+// EmptyPackageImportSet creates a new empty set of PackageImport references
+func EmptyPackageImportSet() *PackageImportSet {
+	return &PackageImportSet{
+		imports: make(map[PackageImport]struct{}),
+	}
+}
+
+// AddImport ensures the set includes the specified import
+// Adding an import already present in the set is fine.
+func (set *PackageImportSet) AddImport(packageImport PackageImport) {
+	set.imports[packageImport] = struct{}{}
+}
+
+// AddReference ensures this set includes an import of the specified reference
+// Adding a reference already in the set is fine.
+func (set *PackageImportSet) AddReference(ref PackageReference) {
```

Comment:
minor: I feel like given the ease of creating an import from a `PackageReference`, and the subtle distinction between the two of them, it would be better to not have this method on this type. It's only used in 1 place and I think it would be clearer if that usage were:
`set.AddImport(NewPackageImport(ref))` rather than `set.AddReference(ref)`.

The second flavoring suggests some "reference-aware storage" in the `set` which isn't the case.

Reply:
Yeah that's also fine